js/components/ReportSummary.vue

```diff
@@ -18,7 +23,19 @@ export default {
 
   props: {
     title: String,
-    totalRecords: Number
+    totalRecords: Number,
+    strategy: String,
+    data: Array
+  },
+
+  data() {
+    return {
+      summaryData: []
+    };
+  },
+
+  mounted() {
+    this.summaryData = countBy(this.data);
```

Comment:
I still find it weird that you can have a prop named `data` and a method named `data` in the same component without one of them being shadowed.

Again, a more natural way of handling this might be to rename the `data` prop to `summaryData` and use a `summaryCounts` computed property that transforms it.